src/components/specific/groups/group-import/GroupImport.vue

```diff
@@ -0,0 +1,211 @@
+                  groups[children.project_id].length === children.list.length
+                "
+              />
+              <span>{{ $t("GroupImport.selectAll") }}</span>
+            </div>
+            <div class="group-import__children-menu__header__divider" />
+          </div>
+        </template>
+        <template #child-item="{ child }">
+          <BIMDataCheckbox
+            class="group-import__children-menu__item--checkbox"
+            margin="0 6px 0 0"
+            :modelValue="
+              Boolean(
+                groups[child.project.id]?.find(item => item.id === child.id)
+              )
+            "
+          />
+          <span>{{ child.text }}</span>
+        </template>
+        <template #child-footer="{ children }">
+          <BIMDataButton
+            class="group-import__children-menu--footer"
+            :disabled="groups[children.project_id].length === 0"
+            color="primary"
+            fill
+            radius
+            width="95%"
+            @click="importGroup(children)"
+          >
+            <span>{{ $t("GroupImport.add") }}</span>
+          </BIMDataButton>
+        </template>
+      </BIMDataMenu>
+    </template>
+  </div>
+</template>
+
+<script>
+import { ref } from "vue";
+import { useGroups } from "../../../../state/groups.js";
+import { useProjects } from "../../../../state/projects.js";
+import { useToggle } from "../../../../composables/toggle.js";
+import GroupService from "../../../../services/GroupService.js";
+
+export default {
+  props: {
+    project: {
+      type: Object,
+      required: true
+    }
+  },
+  setup(props) {
+    const { spaceProjects } = useProjects();
+    const { loadProjectGroups } = useGroups();
+    const { isOpen, close, open } = useToggle();
+
+    const groups = ref({});
+    const projectsToDisplay = ref([]);
+
+    const openGroupImport = async () => {
+      if (isOpen.value) return close();
+
+      projectsToDisplay.value = await Promise.all(
+        spaceProjects.value
+          .filter(({ id }) => id !== props.project.id)
+          .map(async project => {
+            groups.value[project.id] = [];
+
+            let children = {};
+            if (project.isAdmin) {
+              children = {
+                project_id: project.id,
+                list: (await GroupService.fetchProjectGroups(project)).map(
+                  group => {
+                    const currentGroup = {
+                      ...group,
+                      text: group.name,
+                      project,
+                      action: () => checkItem(currentGroup)
+                    };
+                    return currentGroup;
+                  }
+                )
+              };
+            }
+
+            return {
+              ...project,
+              text: project.name,
+              children
+            };
+          })
+      );
+      open();
+    };
+
+    const importGroup = async children => {
```

Comment:
This function looks like something that can be "lifted up" into the state directly.
Let's have a talk about that when you have a moment. ;)